numpy/_core/src/umath/string_ufuncs.cpp

```diff
@@ -345,28 +345,51 @@ string_replace(Buffer<enc> buf1, Buffer<enc> buf2, Buffer<enc> buf3, npy_int64 c
 
     Buffer<enc> end1 = buf1 + len1;
     // Only try to replace if useful.
-    if (len1 >= len2  // Input is big enough to make a match possible.
-        && len2 > 0  // Match string is not empty (so output will be finite).
-        && !(len2 == len3 && buf2.strcmp(buf3) == 0)  // Match and replacement differ.
-        ) {
-        for (npy_int64 time = 0; time < count; time++) {
-            npy_intp pos = findslice_for_replace(buf1, end1 - buf1, buf2, len2);
-            if (pos < 0) {
-                break;
+    if (count > 0        // There's something to replace
```

Comment:
I didn't have `count > 0` originally, because the for loops would trivially not execute. But arguably this is better as the intent is clearer.